osquery/tables/system/windows/wmi.cpp

```diff
@@ -0,0 +1,320 @@
+
+  if (hr != S_OK) {
+    VariantClear(&value);
+    return Status(-1, "Error retrieving data from WMI query.");
+  }
+  if (value.vt != VT_BOOL) {
+    VariantClear(&value);
+    return Status(-1, "Invalid data type returned.");
+  }
+  ret = value.boolVal == VARIANT_TRUE ? true : false;
+  VariantClear(&value);
+  return Status(0);
+}
+
+Status WmiResultItem::GetLong(const std::string& name, long& ret) const {
+  std::wstring property_name = string_to_wstring(name);
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK) {
+    VariantClear(&value);
+    return Status(-1, "Error retrieving data from WMI query.");
+  }
+  if (value.vt != VT_I4) {
+    VariantClear(&value);
+    return Status(-1, "Invalid data type returned.");
+  }
+  ret = value.lVal;
+  VariantClear(&value);
+  return Status(0);
+}
+
+Status WmiResultItem::GetUnsignedLong(const std::string& name,
+                                      unsigned long& ret) const {
+  std::wstring property_name = string_to_wstring(name);
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK) {
+    VariantClear(&value);
+    return Status(-1, "Error retrieving data from WMI query.");
+  }
+  if (value.vt != VT_UI4) {
+    VariantClear(&value);
+    return Status(-1, "Invalid data type returned.");
+  }
+  ret = value.lVal;
+  VariantClear(&value);
+  return Status(0);
+}
+
+Status WmiResultItem::GetLongLong(const std::string& name,
+                                  long long& ret) const {
+  std::wstring property_name = string_to_wstring(name);
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK) {
+    VariantClear(&value);
+    return Status(-1, "Error retrieving data from WMI query.");
+  }
+  if (value.vt != VT_I8) {
+    VariantClear(&value);
+    return Status(-1, "Invalid data type returned.");
+  }
+  ret = value.lVal;
+  VariantClear(&value);
+  return Status(0);
+}
+
+Status WmiResultItem::GetUnsignedLongLong(const std::string& name,
+                                          unsigned long long& ret) const {
+  std::wstring property_name = string_to_wstring(name);
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK) {
+    VariantClear(&value);
+    return Status(-1, "Error retrieving data from WMI query.");
+  }
+  if (value.vt != VT_UI8) {
+    VariantClear(&value);
+    return Status(-1, "Invalid data type returned.");
+  }
+  ret = value.lVal;
+  VariantClear(&value);
+  return Status(0);
+}
+
+Status WmiResultItem::GetString(const std::string& name,
+                                std::string& ret) const {
+  std::wstring property_name = string_to_wstring(name);
+  VARIANT value;
+  HRESULT hr = result_->Get(property_name.c_str(), 0, &value, nullptr, nullptr);
+  if (hr != S_OK) {
+    ret = "";
+    VariantClear(&value);
```

Comment:
We had talked about being too aggressive with the `VariantClear` calls. I made a small test application and running `VariantClear` on a `VARIANT` didn't seem to have any consequences, however reading into the MSDN: `Do not use VariantClear on unitialized variants`